cmake/macros/macro_enable_if_links.cmake

```diff
@@ -23,21 +23,19 @@
 #
 
 MACRO(ENABLE_IF_LINKS _variable _flag)
+  # keep on top to avoid cluttering the _flag and _flag_stripped variables
+  ENABLE_IF_SUPPORTED(CMAKE_REQUIRED_FLAGS "-Werror")
+
   STRING(STRIP "${_flag}" _flag_stripped)
   IF(NOT "${_flag_stripped}" STREQUAL "")
     STRING(REGEX REPLACE "^-" "" _flag_name "${_flag_stripped}")
-    STRING(REPLACE "," "" _flag_name "${_flag_name}")
-    STRING(REPLACE "-" "_" _flag_name "${_flag_name}")
-    STRING(REPLACE "+" "_" _flag_name "${_flag_name}")
-    SET(_backup ${CMAKE_REQUIRED_LIBRARIES})
+    STRING(REGEX REPLACE "\[-+,\]" "_" _flag_name "${_flag_name}")
+
     LIST(APPEND CMAKE_REQUIRED_LIBRARIES "${_flag_stripped}")
-    CHECK_CXX_COMPILER_FLAG(
-      ""
-      DEAL_II_HAVE_FLAG_${_flag_name}
-      )
-    SET(CMAKE_REQUIRED_LIBRARIES ${_backup})
+    CHECK_CXX_SOURCE_COMPILES("int main(){}" DEAL_II_HAVE_LINKER_FLAG_${_flag_name})
```

Comment:
We can no longer use the `CHECK_CXX_COMPILER_FLAG` macro because cmake no longer invokes the linker.